docs/content/ingestion/flatten-json.md

```diff
@@ -0,0 +1,146 @@
+ "dim2": "asdf",
+ "dim3": "zxcv",
+ "ignore_me": "ignore this",
+ "metrica": 9999,
+ "foo": {"bar": "abc"},
+ "foo.bar": "def",
+ "nestmet": {"val": 42},
+ "hello": [1.0, 2.0, 3.0, 4.0, 5.0],
+ "world": [{"hey": "there"}, {"tree": "apple"}],
+ "thing": {"food": ["sandwich", "pizza"]}
+}
+```
+
+The column "metrica" is a Long metric column, "hello" is an array of Double metrics, and "nestmet.val" is a nested Long metric. All other columns are dimensions.
+
+To flatten this JSON, the parseSpec could be defined as follows:
+
+```json
+"parseSpec": {
+  "format": "json",
+  "flattenSpec": {
+    "enabled": true,
+    "useFieldDiscovery": true,
+    "fields": [
+      {
+        "type": "root",
+        "name": "dim1",
+        "expr": "dim1"
+      },
+      "dim2",
+      {
+        "type": "nested",
+        "name": "foo.bar",
+        "expr": "$.foo.bar"
+      },
+      {
+        "type": "root",
+        "name": "root-foo.bar",
+        "expr": "foo.bar"
+      },
+      {
+        "type": "nested",
+        "name": "nested-metric",
+        "expr": "$.nestmet.val"
+      },
+      {
+        "type": "nested",
+        "name": "hello-0",
+        "expr": "$.hello[0]"
+      },
+      {
+        "type": "nested",
+        "name": "hello-4",
+        "expr": "$.hello[4]"
+      },
+      {
+        "type": "nested",
+        "name": "world-hey",
+        "expr": "$.world.hey"
+      },
+      {
+        "type": "nested",
+        "name": "worldtree",
+        "expr": "$.world.tree"
+      },
+      {
+        "type": "nested",
+        "name": "first-food",
+        "expr": "$.thing.food[0]"
+      },
+      {
+        "type": "nested",
+        "name": "second-food",
+        "expr": "$.thing.food[1]"
+      }
+    ]
+  },
+  "dimensionsSpec" : {
+   "dimensions" : [],
+   "dimensionsExclusions": ["ignore_me"]
+  },
+  "timestampSpec" : {
+   "format" : "auto",
+   "column" : "timestamp"
+  }
+}
+```
+
+Fields "dim3", "ignore_me", and "metrica" will be automatically discovered because 'useFieldDiscovery' is true, so they have been omitted from the field spec list.
+
+Aggregators should use the metric column names as defined in the flattenSpec. Using the example above:
+
+```json
+"metricsSpec" : [ 
+{
+  "type" : "longSum",
+  "name" : "Nested Metric Value",
+  "fieldName" : "nested-metric"
+}, 
+{
+  "type" : "doubleSum",
+  "name" : "Hello Index #0",
+  "fieldName" : "hello-0"
+},
+{
+  "type" : "longSum",
+  "name" : "metrica",
+  "fieldName" : "metrica"
+}
+]
+```
+
+Note that:
+
+* For convenience, when defining a root-level field, it is possible to define only the field name, as a string, shown with "dim2" above.
+* Enabling 'useFieldDiscovery' will only autodetect fields at the root level with a single value, not a map or list. In the example above, "dim1", "dim2", "dim3", "ignore_me", "metrica", and "foo.bar" (at the root) would be automatically detected as columns, but the fields "hello" and "world" must be explicitly defined.  
```

Comment:
this should detect fields that are Lists of simple types, so please update the docs along with the parser.